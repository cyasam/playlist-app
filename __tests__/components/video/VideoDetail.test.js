import React from 'react'
import { shallow } from 'enzyme'
import VideoDetail from '../../../src/scripts/components/video/VideoDetail'

describe('VideoDetail Component', () => {
  const setup = (propOverrides) => {
    const props = {
      videoId: 'g884wLr9gNc', ...propOverrides
    }
    const wrapper = shallow(<VideoDetail { ...props } />)
    return {
      wrapper,
      props
    }
  }

  it('renders properly', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })
  
  it('renders properly if videos prop is empty', () => {
    const { wrapper } = setup({ videoId: '' })
    expect(wrapper).toMatchSnapshot()

    expect(wrapper.find('iframe').exists()).toBe(false)
  })
})