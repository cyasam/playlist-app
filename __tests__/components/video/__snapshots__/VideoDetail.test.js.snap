// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoDetail Component renders properly 1`] = `
<div>
  <iframe
    allow="encrypted-media"
    allowFullScreen={true}
    frameBorder="0"
    height="315"
    src="https://www.youtube.com/embed/g884wLr9gNc?autoplay=1"
    width="560"
  />
</div>
`;

exports[`VideoDetail Component renders properly if videos prop is empty 1`] = `
<div>
  Error: Video not found.
</div>
`;
